prime-angular-frontend/src/app/modules/adjudication/pages/health-authorities/health-auth-care-types-page/health-auth-care-types-page.component.ts

```diff
@@ -52,20 +60,33 @@ export class HealthAuthCareTypesPageComponent implements OnInit {
 
   public onSubmit() {
     if (this.formUtilsService.checkValidity(this.form)) {
-      const careTypes = [...new Set(this.careTypes.value.map(({ careType }) => careType.trim()) as string[])];
+      const careTypes = [...new Set(this.careTypes.getRawValue()
+        .map(({ careType }) => careType.name.trim()) as string[])];
       this.healthAuthResource.updateHealthAuthorityCareTypes(this.route.snapshot.params.haid, careTypes)
         .subscribe(() => this.nextRouteAfterSubmit());
     }
   }
 
-  public addCareType(careType: string = '') {
+  public addCareType(careType: HealthAuthorityCareTypeMap = null) {
     this.careTypes.push(this.fb.group({
-      careType: [careType ?? '', Validators.required]
+      careType: [{ value: careType, disabled: careType?.id }, Validators.required]
     }));
   }
 
   public removeCareType(index: number) {
-    this.careTypes.removeAt(index);
+    const careTypeId = this.careTypes.value[index]?.careType?.id;
+    if (careTypeId) {
+      this.healthAuthResource.getHealthAuthorityCareTypeSiteIds(this.route.snapshot.params.haid, careTypeId)
+        .subscribe((healthAuthoritySites) => {
+          (!healthAuthoritySites.length)
+            ? this.careTypes.removeAt(index)
+            : this.toastService.openErrorToast('Care type could not be removed, one or more sites are using it');
+        });
+    } else {
+      // when careTypeId is undefined that means we're deleting a care type after adding it and before hitting Save and Continue
+      // i.e. before it was given an Id
+      this.careTypes.removeAt(index);
```

Comment:
In cases like this when we have an if else and one side is a lot simpler of logic we can do something like this
```
public removeCareType(index: number) {
   const careTypeId = this.careTypes.value[index]?.careType?.id;
   if (!careTypeId) {
      this.careTypes.removeAt(index);
      return;
   }
   ... rest
}
```

This helps in the long run to keep indentation as clean as possible since the api call is no longer wrapped in an if/else.

Also i don't think this comment adds to the code.

Reply:
Fixed for both careTypes and vendors